sdk/storage/storage-file/src/DirectoryClient.ts

```diff
@@ -556,7 +581,7 @@ export class DirectoryClient extends StorageClient {
    * @returns {AsyncIterableIterator<Models.DirectoryListFilesAndDirectoriesSegmentResponse>}
    * @memberof DirectoryClient
    */
-  private async *listSegments(
+  private async *iterateFilesAndDirectoriesSegments(
```

Comment:
Renaming them to avoid confusion between `listSegments`, `listHandlesSegment`, etc.